qiskit_machine_learning/algorithms/effective_dimension.py

```diff
@@ -0,0 +1,337 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""An implementation of the effective dimension algorithm."""
+
+import time
+from typing import Optional, Union, List, Callable, Tuple
+
+import numpy as np
+from scipy.special import logsumexp
+
+from ..neural_networks import OpflowQNN, NeuralNetwork
+from ..exceptions import QiskitMachineLearningError
+
+
+class EffectiveDimension:
+
+    """
+    This class computes the global effective dimension for Qiskit NeuralNetworks following the algorithm
+    presented in "The Power of Quantum Neural Networks": https://arxiv.org/abs/2011.00027.
+    """
+
+    def __init__(
+        self,
+        qnn: NeuralNetwork,
+        params: Optional[Union[List[float], np.ndarray, float]] = None,
+        inputs: Optional[Union[List[float], np.ndarray, float]] = None,
+        num_params: Optional[int] = 1,
+        num_inputs: Optional[int] = 1,
+        fix_seed=False,
```

Comment:
Unless there is good reason to have a different seed on each instance we could go with the algorithm globals random number generator. When seeded that gives a reproducible sequence, otherwise by default its seeded randomly.